lib/src/otx/data/dataset/base_new.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Base class for OTXDataset using new Datumaro experimental Dataset."""
+
+from __future__ import annotations
+
+import abc
+from typing import TYPE_CHECKING, Callable, Iterable, List, Union
+
+import numpy as np
+import torch
+from torch.utils.data import Dataset as TorchDataset
+
+if TYPE_CHECKING:
+    from datumaro.experimental import Dataset
+
+from otx.data.entity.sample import OTXSample
+from otx.data.entity.torch.torch import OTXDataBatch
+from otx.data.transform_libs.torchvision import Compose
+from otx.types.image import ImageColorChannel
+
+Transforms = Union[Compose, Callable, List[Callable], dict[str, Compose | Callable | List[Callable]]]
+
+
+def _default_collate_fn(items: list[OTXSample]) -> OTXDataBatch:
+    """Collate OTXSample items into an OTXDataBatch.
+
+    Args:
+        items: List of OTXSample items to batch
+    Returns:
+        Batched OTXSample items with stacked tensors
+    """
+    # Convert images to float32 tensors before stacking
+    image_tensors = []
+    for item in items:
+        img = item.image
+        if isinstance(img, torch.Tensor):
+            # Convert to float32 if not already
+            if img.dtype != torch.float32:
+                img = img.float()
+        else:
+            # Convert numpy array to float32 tensor
+            img = torch.from_numpy(img).float()
+        image_tensors.append(img)
+
+    # Try to stack images if they have the same shape
+    if len(image_tensors) > 0 and all(t.shape == image_tensors[0].shape for t in image_tensors):
+        images = torch.stack(image_tensors)
+    else:
+        images = image_tensors
+
+    return OTXDataBatch(
+        batch_size=len(items),
+        images=images,
+        labels=[item.label for item in items] if items[0].label is not None else None,
+        masks=[item.masks for item in items] if any(item.masks is not None for item in items) else None,
+        bboxes=[item.bboxes for item in items] if any(item.bboxes is not None for item in items) else None,
+        keypoints=[item.keypoints for item in items] if any(item.keypoints is not None for item in items) else None,
+        polygons=[item.polygons for item in items if item.polygons is not None]
+        if any(item.polygons is not None for item in items)
+        else None,
+        imgs_info=[item.img_info for item in items] if any(item.img_info is not None for item in items) else None,
+    )
+
+
+class OTXDataset(TorchDataset):
```

Comment:
If the old and new `OTXDataset` have the same API, I suggest to add some black-box functional tests that instantiate both classes, apply a sequence of operations to each, then finally assert that the respective results are identical.